server/sonar-server/src/main/java/org/sonar/server/rule/RuleUpdate.java

```diff
@@ -212,27 +223,37 @@ private boolean isCustomRuleFieldsEmpty() {
   }
 
   private void checkCustomRule() {
-    if (!isCustomRule) {
+    if (useCase != CUSTOM_RULE) {
       throw new IllegalStateException("Not a custom rule");
     }
   }
 
+  public OrganizationDto getOrganization() {
+    return organization;
+  }
+
   /**
    * Use to update a rule provided by a plugin (name, description, severity, status and parameters cannot by changed)
    */
   public static RuleUpdate createForPluginRule(RuleKey ruleKey) {
-    RuleUpdate ruleUpdate = new RuleUpdate(ruleKey);
-    ruleUpdate.isCustomRule = false;
-    return ruleUpdate;
+    return new RuleUpdate(ruleKey, PLUGIN_RULE);
   }
 
   /**
    * Use to update a custom rule
    */
   public static RuleUpdate createForCustomRule(RuleKey ruleKey) {
-    RuleUpdate ruleUpdate = new RuleUpdate(ruleKey);
-    ruleUpdate.isCustomRule = true;
-    return ruleUpdate;
+    return new RuleUpdate(ruleKey, CUSTOM_RULE);
   }
 
+  public enum RuleUpdateUseCase {
```

Comment:
I don't see the point to introduce this enum compared to boolean `isCustom`, but ok...